internal/gengapic/auxiliary_test.go

```diff
@@ -0,0 +1,447 @@
+				},
+			},
+			want: map[string]operationWrapper{
+				"CreateFooOperation": {
+					name: "CreateFooOperation",
+					response: &descriptorpb.DescriptorProto{
+						Name: proto.String("Bar"),
+					},
+					metadata: &descriptorpb.DescriptorProto{
+						Name: proto.String("FooMetadata"),
+					},
+					responseName: protoreflect.FullName("google.other.v1.Bar"),
+					metadataName: protoreflect.FullName("google.other.v1.FooMetadata"),
+				},
+			},
+			wantErrStr: "duplicate operation wrapper types",
+			res:        "Foo",
+			meta:       "FooMetadata",
+			oiRes:      "Foo",
+			oiMeta:     "FooMetadata",
+			m:          "CreateFoo",
+			pkg:        "google.example.v1",
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			r := &descriptorpb.DescriptorProto{
+				Name: proto.String(tC.res),
+			}
+			md := &descriptorpb.DescriptorProto{
+				Name: proto.String(tC.meta),
+			}
+			oi := &longrunningpb.OperationInfo{
+				ResponseType: tC.oiRes,
+				MetadataType: tC.oiMeta,
+			}
+			m := &descriptorpb.MethodDescriptorProto{
+				Name:    proto.String(tC.m),
+				Options: &descriptorpb.MethodOptions{},
+			}
+			proto.SetExtension(m.GetOptions(), longrunningpb.E_OperationInfo, oi)
+
+			f := &descriptorpb.FileDescriptorProto{
+				Package: proto.String(tC.pkg),
+			}
+			messageParent := f
+			if tC.otherPkg != "" {
+				of := &descriptorpb.FileDescriptorProto{
+					Package: proto.String(tC.otherPkg),
+				}
+				messageParent = of
+			}
+
+			g := &generator{
+				descInfo: pbinfo.Info{
+					Type: map[string]pbinfo.ProtoType{
+						fmt.Sprintf(".%s.%s", messageParent.GetPackage(), r.GetName()):  r,
+						fmt.Sprintf(".%s.%s", messageParent.GetPackage(), md.GetName()): md,
+					},
+					ParentFile: map[protoreflect.ProtoMessage]*descriptorpb.FileDescriptorProto{
+						m:  f,
+						r:  messageParent,
+						md: messageParent,
+					},
+				},
+				aux: &auxTypes{
+					opWrappers:      tC.existing,
+					methodToWrapper: map[*descriptorpb.MethodDescriptorProto]operationWrapper{},
+				},
+			}
+
+			err := g.maybeAddOperationWrapper(m)
+			if tC.wantErrStr == "" && err != nil {
+				t.Errorf("maybeAddOperationWrapper(%+v): unexpected error %v", m, err)
+			} else if tC.wantErrStr != "" && err == nil {
+				t.Errorf("maybeAddOperationWrapper(%+v): got no error, want %q, with existing %+v", m, tC.wantErrStr, tC.existing)
+			} else if tC.wantErrStr != "" && err != nil && !strings.Contains(err.Error(), tC.wantErrStr) {
+				t.Errorf("maybeAddOperationWrapper(%+v): got %q, want %q, with existing %+v", m, err.Error(), tC.wantErrStr, tC.existing)
+			} else if diff := cmp.Diff(g.aux.opWrappers, tC.want, cmp.Comparer(proto.Equal), cmp.AllowUnexported(operationWrapper{})); diff != "" {
+				t.Errorf("maybeAddOperationWrapper(%+v): got(-),want(+)\n%s", m, diff)
+			}
+		})
+	}
+}
+
+func TestGenOperations(t *testing.T) {
+	wrappers := map[string]operationWrapper{
```

Comment:
Is it possible to test the sorting in `genOperations`? Would it work to reverse the order of `wrappers` here?

Reply:
Did you one better, I moved the sorting code into their own functions and added tests.